in_toto/user_settings.py

```diff
@@ -76,18 +64,47 @@ def _colon_split(value):
 
 
 def get_env():
-  """Return a dict of environment variables starting with `ENV_PREFIX`
+  """
+  <Purpose>
+    Parse environment for variables with prefix `ENV_PREFIX` and return
+    a dict of key-value pairs.
+
+    The prefix `ENV_PREFIX` is stripped from the keys in the returned dict.
+
+    Values that contain colons (:) are split at the postion of the colons and
+    converted into a list.
+
+
+    Example:
+
+    ```
+    # Exporting variables in e.g. bash
+    export IN_TOTO_ARTIFACT_BASE_PATH='/home/user/project'
+    export IN_TOTO_ARTIFACT_EXCLUDES='*.link:.gitignore'
+    ```
+
+    produces
+
+    ```
+    {
+      "ARTIFACT_BASE_PATH": "/home/user/project"
+      "ARTIFACT_EXCLUDES": ["*.link", ".gitignore"]
+    }
+    ```
+
+  <Exceptions>
+    None.
+
+  <Side Effects>
+    Calls function to read files from disk.
```

Comment:
For a while I used to list similar entries in Side Effects, but others have pointed out that this isn't really a side effect.

Reply:
I remember having this discussion and also that there were uncertainties about some things and whether they would qualify as side effect. But I am pretty sure that there was no doubt about I/O.